### PR TITLE
[FEATURE] - Storage Limit - Reduce Txs Being Stored

### LINKED ISSUE #2914: Alleviate storage limit issue (Internal Ticket)
### **Description**
In the mobile application we are starting to see a Storage Limit Error becoming a more frequent reported issue. This requires the user to setup their wallet each time this error is thrown. The evidence seems to suggest that since we are saving the entire redux state for the app and decrypting it upon login it opens up the potential for data corruption causing the error.

Focus:
1. Store the opt-in to metrics in a different location to allow this issue to be logged.
2. Limit the amount of user generated data that can be stored (e.g. Txs, NFTs). The extension only stores the latest 100 Txs across all accounts.

### **Technical Requirements**
- Pull at least the `analytics` field from redux persist into it's own key so that it's not nested under the `root` key. Hint: Look at the analytics reducer, that's where `analytics` is being set. This addresses focus number one.
- Regarding limiting TXs, limit/truncate the incoming TXs count to 40. This addresses focus number two.
- Will most likely need to update analytics reducer.
- May have to update tests related to redux
- Add to `controller` addTransaction() see extension reference 

### **Acceptance Criteria**
- Ensure that the `analytics` field that controls MetaMetrics is stored outside of the `root` key in redux-persist. 
- Somehow emulate the corrupt storage scenario and verify that events are still being tracked
- Ensure that the max number of TXs cached is 40.
- Still be able to view transactions on different accounts

### **References**
- Proposal of issue & potential solutions - https://consensys.slack.com/archives/GBW7S9FSN/p1626126984211600
- Original issue - https://app.zenhub.com/workspaces/metamask-mobile-5f984938ddc0e4001d4b79cb/issues/metamask/metamask-mobile/2652
- Extension Code: tx-state-manager.js - line 192
  -   // checks if the length of the tx history is longer then desired persistence
      // limit and then if it is removes the oldest confirmed or rejected tx.
      // Pending or unapproved transactions will not be removed by this
      // operation. For safety of presenting a fully functional transaction UI
      // representation, this function will not break apart transactions with the
      // same nonce, per network. Not accounting for transactions of the same
      // nonce and network combo can result in confusing or broken experiences
      // in the UI.

### PR DESCRIPTION
**Pull request by sethkfman**
**Description**

- Limits the numbers of transactions to be displayed and stored to 40 most recent tx
- Added option to view all associated txs block explorer at the footer the list
- Store METRICS_OPTIN & ONBOARDING in using `DefaultPreference` and removed them for redux persist store

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

**Issue**

Resolves #2914

### CODE REVIEW

**Comment by sethkfman** [comment]
@jakehaugen Would you be able to take a look at this I don't if we need to do more design work on it? The view on block explore will be displayed at the bottom of the transaction history & transactions lists.
Screen Shot 2021-08-03 at 1 17 20 PM

File: app/components/UI/Transactions/index.js [hunk 1]
```diff
@@ -59,14 +74,26 @@ class Transactions extends PureComponent {
...
+		/**
+		 * Frequent RPC list from PreferencesController
+		 */
+		frequentRpcList: PropTypes.array,
 		/**
 		/* navigation object required to push new views
 		*/
 		navigation: PropTypes.object,
+		/**
+		 * Object representing the selected the selected network
```
Comment (gantunesr) [inline app/components/UI/Transactions/index.js:94]: Small comment error: `the selected` is repeated

**Comment by jakehaugen** [comment]
> @jakehaugen Would you be able to take a look at this I don't if we need to do more design work on it? The view on block explore will be displayed at the bottom of the transaction history & transactions lists.
> Screen Shot 2021-08-03 at 1 17 20 PM

This looks pretty good to me. Only suggestion is to update the copy to say "View full history on Etherscan". Does that make sense to you?

**Review by gantunesr — APPROVED** [review]
Code looks good and the behaviour is the expected at the extend of the use cases I tested. I'm not sure if there's more changes to be added, so let me know if that is the case.

**Comment by sethkfman** [comment]
@jakehaugen good suggestion. See updated view:
Screen Shot 2021-08-05 at 9 04 10 AM

File: app/components/UI/Transactions/index.js (hunk 1 above)
Comment (sethkfman) [reply app/components/UI/Transactions/index.js]: Got it. Thanks

**Review by ibrahimtaveras00 — APPROVED** [review]
QA Passed 👍🏽

**Pull request merged**